reps/2023-04-28-remove-algorithms-from-rllib.md

```diff
@@ -0,0 +1,193 @@
+
+We will generally accept contributions to this repo that meet any of the following criteria:
+1. Updating dependencies.
+2. Submitting community contributed algorithms that have been tested and are ready for use.
+3. Enabling algorithms to be run in different environments (ex. adding support for a new type of gym environment).
+4. Updating algorithms for use with the newer RLlib API.
+5. General bug fixes.
+
+We will not accept contributions that generally add significant new maintenance burden.
+
+### Contributing new algorithms
+If you would like to contribute a new algorithm to this repo, please follow the following steps:
+1. Create a new directory with the same structure as the other algorithms
+2. Add a `README.md` file that describes the algorithm and its usecases
+3. Create tests for it that run in under 10 minutes on a single machine (gh actions runners are 2 core machines). Then create a script that can be used as a long running benchmark for your algorithm.
+4. Submit a PR and a RLlib maintainer will review it and help you set up your testing to integrate with the CI of this repo.
+
+
+### Telemetry and Promoting Algorithms to RLlib
+
+In Ray we have telemetry features that allow us to track the usage of algorithms. We'll have to establish a similar
+telemetry system for this repo. If an algorithm shows considerable ussage compared to the ussage that we see in RLlib, we can consider promoting it to RLlib and moving the maintenance burden to the RLlib team.
+
+
+### Testing
+
+Testing will be done with github actions. Each algorithm will have its own github action workflow that will run the unit tests under the algorithm's `tests/` directory. A separate workflow will be created for every algorithm that allows for a long running test to be launched on the Anyscale platform using Anyscale jobs. This test is mainly for performance checking and can only be triggered by a code owner. Testing is only triggered on pull requests. This is because we don't expect the dependencies of the library to change / not be hard-pinned and to reduce the maintenance of the ci.
+
+
+## Compatibility, Deprecation, and Migration Plan
+
+### Compatibility
+The dependencies of each algorithm are hard pinned for each algorithm's package. This means that if a user wants to use a newer version of ray, the burden of migrating the algorithm to the new version of ray is on the user.
+
+### Deprecation
+1. We will keep the deprecated algorithms' classes around in the repo for at least 3 releases. We will add standard deprecation warnings to the algorithms and their components, but after the 3 releases we will add deprecation errors
+any time an algorithm or its related policy/model components are used.
+
+2. We will add thorough documentation to https://docs.ray.io/en/latest/rllib/rllib-algorithms.html that explains why the algorithms were deprecated and how to migrate to the new rllib contrib algorithm python packages.
```

Comment:
We should just re-link the algorithm docs we had to external docs on rllib-contrib.